tests/common.go

```diff
@@ -7725,3 +7777,183 @@ func GetPoolCapacityUsed(poolUUID string) (float64, error) {
+			if err != nil {
+				return err
+			}
+
+			//waiting for volumes replication status should be up
+			for _, v := range appVolumes {
+				t := func() (interface{}, bool, error) {
+					replicationStatus, err := GetVolumeReplicationStatus(v)
+					if err != nil {
+						return nil, false, err
+					}
+
+					if replicationStatus == "Up" {
+						return "", false, nil
+					}
+					if replicationStatus == "Resync" {
+						return "", true, fmt.Errorf("volume %s is still in Resync state", v.ID)
+					}
+					return "", false, fmt.Errorf("volume %s is in %s state cannot proceed further", v.ID, replicationStatus)
+				}
+				_, err = task.DoRetryWithTimeout(t, 60*time.Minute, 1*time.Minute)
+				if err != nil {
+					return err
+				}
+			}
+
+			log.InfoD(fmt.Sprintf("scale down app %s to 0", ctx.App.Key))
+			applicationScaleMap, err := Inst().S.GetScaleFactorMap(ctx)
+			if err != nil {
+				return err
+			}
+
+			applicationScaleDownMap := make(map[string]int32, len(ctx.App.SpecList))
+
+			for name, _ := range applicationScaleMap {
+				applicationScaleDownMap[name] = 0
+			}
+			err = Inst().S.ScaleApplication(ctx, applicationScaleDownMap)
+			if err != nil {
+				return err
+			}
+			//waiting for volumes to be detached after scale down
+			for _, v := range appVolumes {
+				t := func() (interface{}, bool, error) {
+					apiVol, err := Inst().V.InspectVolume(v.ID)
+					if err != nil {
+						return "", false, err
+					}
+					if len(apiVol.AttachedOn) == 0 {
+						return "", false, nil
+					}
+					return "", true, fmt.Errorf("volume %s is still attached to %s", v.ID, apiVol.AttachedOn)
+				}
+				_, err = task.DoRetryWithTimeout(t, waitResourceCleanup, 10*time.Second)
+				if err != nil {
+					return err
+				}
+			}
+
+			for _, v := range appVolumes {
+				replicaSets, err := Inst().V.GetReplicaSets(v)
+				if err != nil {
+					return err
+				}
+
+				var checksum string
+				var primaryPool string
+				var primaryNode *node.Node
+				for _, rs := range replicaSets {
+					poolUuids := rs.PoolUuids
+					for _, poolUuid := range poolUuids {
+						nodeDetail, err := GetNodeWithGivenPoolID(poolUuid)
+						if err != nil {
+							return err
+						}
+						poolID, err := GetPoolIDFromPoolUUID(poolUuid)
+						if err != nil {
+							return err
+						}
+
+						inspectVolume, err := Inst().V.InspectVolume(v.ID)
+						if err != nil {
+							return err
+						}
+						log.Infof("Getting md5sum for volume %s on pool %s in node %s", inspectVolume.Id, poolID, nodeDetail.Name)
+						//To-Do update the command if set up is dmthin
+						cmd := fmt.Sprintf("/opt/pwx/bin/runc exec -t portworx md5sum /var/.px/%d/%s/pxdev", poolID, inspectVolume.Id)
+						output, err := Inst().N.RunCommand(*nodeDetail, cmd, node.ConnectionOpts{
+							Timeout:         defaultTimeout,
+							TimeBeforeRetry: defaultRetryInterval,
+							Sudo:            true,
+						})
+						if err != nil {
+							return err
+						}
```

Comment:
Can you test this on aggregated volume as well

Reply:
I have separate PR in-progress for aggregated volumes, will test this as part of that